packages/core/src/build/configAndIndexingFunctions.ts

```diff
@@ -322,32 +322,32 @@ export async function buildConfigAndIndexingFunctions({
         registeredFunctionSelectors.push(abiFunction.selector);
       }
 
-      let topic0: LogTopic = registeredEventSelectors;
-      let topic1: LogTopic = null;
-      let topic2: LogTopic = null;
-      let topic3: LogTopic = null;
+      let topicsArray: {
+        topic0: LogTopic;
+        topic1: LogTopic;
+        topic2: LogTopic;
+        topic3: LogTopic;
+      }[] = [
+        {
+          topic0: registeredEventSelectors,
+          topic1: null,
+          topic2: null,
+          topic3: null,
+        },
+      ];
 
       if (source.filter !== undefined) {
```

Comment:
I'm finding this code block a little hard to follow. I think you have it right: multiple filters in the config should result to multiple sources. Might be helpful to try to limit the use of `.reduce`